docs/reference/mapping/fields/synthetic-source.asciidoc

```diff
@@ -0,0 +1,170 @@
+[[synthetic-source]]
+==== Synthetic `_source`
+
+Though very handy to have around, the source field takes up a significant amount
+of space on disk. Instead of storing source documents on disk exactly as you
+send them, Elasticsearch can reconstruct source content on the fly. Enable this
+by setting `synthetic: true` in `_source`:
+
+[source,console,id=enable-synthetic-source-example]
+----
+PUT idx
+{
+  "mappings": {
+    "_source": {
+      "synthetic": true
+    }
+  }
+}
+----
+// TESTSETUP
+
+While this on the fly reconstruction is *generally* slower than saving the source
```

Comment:
it is slower at search time but doesn't it make indexing faster? Or is the indexing speed gain negligible?

Reply:
It's complicated. We still make `_recovery_source` so we still have to write the source, but we remove that `_recovery_source` pretty darn quick. I don't expect really much indexing performance change, but merging should be marginally faster.